create_database/pubtator-to-postgres.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Inserting Pubtator into a Database
+
+# This notebook is designed to load each [Pubmed](https://www.ncbi.nlm.nih.gov/pubmed/) abstract. It uses our own [pubtator repository](https://github.com/greenelab/pubtator) to convert each abstract into the appropriate format (xml) to load into a postgres database. Run the pubtator scripts before running this notebook, so the pubtator xml file can be constructed. Once constructed this notebook is designed to parse the data into a database. After loading each abstract, candidate extraction is performed for each relationship type.
+
+# ## MUST RUN AT THE START OF EVERYTHING
+
+# Load all necessary imports for the rest of this notebook. Plus, set up the postgres database for database operations. 
+
+# In[ ]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+#Imports
+import csv
+import os
+import random
+
+import numpy as np
+import pandas as pd
+import tqdm
+
+
+# In[ ]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[ ]:
+
+
+from snorkel.candidates import PretaggedCandidateExtractor
+from snorkel.models import Document, Sentence, candidate_subclass
+from snorkel.parser import CorpusParser
+from snorkel.viewer import SentenceNgramViewer
+from snorkel.parser import DocPreprocessor
+from snorkel.models import Document
+
+from sqlalchemy import func
+from string import punctuation
+import lxml.etree as et
+
+
+# In[ ]:
+
+
+def offsets_to_token(left, right, offset_array, lemmas, punc=set(punctuation)):
```

Comment:
Would it make more sense to have these scripts be outside of the nb?

Reply:
Yes I agree. I'll move the functions into a separate script.